src/Microsoft.OData.Core/ODataLibraryCompatibility.cs

```diff
@@ -4,29 +4,54 @@
 // </copyright>
 //---------------------------------------------------------------------
 
-using System.Diagnostics.CodeAnalysis;
+using System;
 
 namespace Microsoft.OData
 {
     /// <summary>
     /// Library compatibility levels.
     /// </summary>
-    [SuppressMessage("Microsoft.Design", "CA1008:EnumsShouldHaveZeroValue")]
-    public enum ODataLibraryCompatibility
+    [Flags]
+    public enum ODataLibraryCompatibility : long
     {
+        /// <summary>
+        /// No specific version or feature set.
+        /// </summary>
+        None = 0, // No flags set
+
+        /// <summary>
+        /// When enabled, the CSDL Scale and SRID "variable" value will be written "Variable" instead of "variable" for compatibility with older versions of the library.
+        /// </summary>
+        UseLegacyVariableCasing = 1 << 0,
+
+        /// <summary>
+        /// When enabled, writes a null odata annotation.@odata.null=true
+        /// </summary>
+        WriteNullODataAnnotation = 1 << 1,
+
+        /// <summary>
+        /// When enabled, writes a context annotation for a navigation property.
+        /// </summary>
+        WriteODataContextAnnotationForNavProperty = 1 << 2,
+
+        /// <summary>
+        /// When enabled, writes top level properties.
+        /// </summary>
+        WriteTopLevelProperty = 1 << 3,
+
         /// <summary>
         /// Version 6.x
         /// </summary>
-        Version6 = 060000,
+        Version6 = UseLegacyVariableCasing | WriteNullODataAnnotation | WriteODataContextAnnotationForNavProperty,

```

Comment:
Looking at the usage, `WriteNullODataAnnotation` is being checked where in the old code `Version7` was checked, but here the flag is set for `Version6` and not `Version7`. I find this confusing. Shouldn't it be set of Version7 and Latest instead? Also, shouldn't this be the default behaviour in newer versions? I think the intent of the compatibility flags was to optional activate the behaviour of old versions when we change the default behaviour. If something is enabled in the last version, then it should be the default behaviour and therefore does not need a flag. If we want the user to be able to disable the new default behaviour for better compatibility with older versions, then I think we should consider adding a "negative" flag instead that optional disables the new behaviour. For example, adding a `DoNotWriteNullODataAnnotation` flag that is only set in V6.

Reply:
If we don't use the "negative flag" approach, the we should use the "new behaviour flags" for all the versions where that new behavior is applicable (including Latest)

Reply:
It was if less than Version7 not Version7 which means it was meant for Version6

Reply:
We write a null OData Annotation in 6 not in the latest versions.